$NetBSD: patch-ab,v 1.1.2.2 2002/06/23 19:03:01 jlam Exp $

the time stuff is needed to compile on alpha, the other stuff kills
compiler warnings

--- MultiClock.c.orig	Sat Jun 15 16:09:44 2002
+++ MultiClock.c
@@ -33,7 +33,7 @@
 #include <X11/Xaw/XawInit.h>
 #include "MultiClockP.h"
 
-extern long time();
+#include <time.h>
 static void clock_tic(), updateClocks(), DrawHand(), DrawSecond(), SetSeg(), DrawClockFace();
 static erase_hands(), round();
 	
@@ -177,7 +177,7 @@
     {
       ClassField(numberOfSlots) += CLOCK_LIST_SIZE;
       ClassField(clockList) =
-	(ClockWidget *) XtRealloc(ClassField(clockList),
+	(ClockWidget *) XtRealloc((void *) ClassField(clockList),
 				  sizeof(ClockWidget)*ClassField(numberOfSlots));
     }
 
@@ -239,7 +239,7 @@
     if(!w->clock.analog) {
        char *str;
        struct tm tm, *localtime();
-       long time_value;
+       time_t time_value;
        (void) time(&time_value);
        tm = *localtime(&time_value);
        str = asctime(&tm);
@@ -391,7 +391,7 @@
         XtIntervalId *id;
 {
   ClockWidget w = (ClockWidget) client_data;
-  long	time_value;
+  time_t	time_value;
   char localTimezone[TZ_BUFLEN];
   char *tzenv = getenv("TZ");
   struct tm *localtime();
@@ -417,7 +417,7 @@
     {
       (*list)->clock.show_second_hand =
 	(ClassField(updateInterval) <= SECOND_HAND_TIME);
-      if (XtIsRealized(*list))
+      if (XtIsRealized((Widget) *list))
 	{
 	  if ((*list)->clock.timezone)
 	  {
